### ISSUE TITLE
MaxListenersExceededWarning

### ISSUE THREAD

**Issue by VikR0001** [GraphQLCollege/graphql-postgres-subscriptions#10]
I'm seeing this in my server logs:

> MaxListenersExceededWarning: Possible EventEmitter memory leak detected. 11 CallAdded listeners added.

Question:

- Is there a limit of 10 listeners per client? If so, my app only has 4 subscriptions. How can I have 10 listeners? 
- Is there a limit of 10 listeners for the entire app? If so how can this function if I have hundreds of simultaneous users?

I'm happy to correct this, and need a pointer in the right direction. :)

**Comment by JulianMayorga** [comment]
Hi @VikR0001, thanks so much for contributing. I'll try to see if there's a memory leak somewhere in the code.

Knowing a bit more about your usage of the library would be a great step towards understanding the issue. Would you be so kind to write a test case that reproduces the problem you're having? Like writing a test that mimics the way you use the library and check if there's a `MaxListenersExceededWarning` error? That would be a huge help.

**Comment by VikR0001** [comment]
I would love to. Is there a demo project on Codepen or on a similar site that I can start from?

**Comment by JulianMayorga** [comment]
Sure thing! You have two options:

1. You can clone this project and add a new test case. You'd need docker and docker compose to run the test suite with `docker-compose run build` and then `docker-compose run test`.
2. Cloning this [Glitch example](https://glitch.com/~pinapp-subscriptions). You can read more about this example in [Fullstack GraphQL's subscribtion chapter](https://github.com/GraphQLCollege/fullstack-graphql/blob/master/manuscript/chapter-5.md).

Reproducing the error in either option would be amazing. Thanks so much @VikR0001!

**Comment by VikR0001** [comment]
I thought I'd create a test case app by just cutting everything out of the app I'm developing that wasn't needed for the test case. But-- it appears I'm not getting `MaxListenersExceededWarning` any more. Possibly when I unsubscribed to my graphQL subscription on component unmount that fixed it-- but I still got the `MaxListenersExceededWarning` warnings until I did a server reboot. 

Anyway, so far things appear to be all good. If the anomaly recurs I will make a test case app and post a link to it here.
